src/utils/waitForAll.ts

```diff
@@ -4,11 +4,15 @@ import { createMemoizeAtom } from './weakCache'
 
 const memoizeAtom = createMemoizeAtom()
 
-type ResolveType<T> = T extends Promise<infer V> ? V : T
+type ResolvePromise<T> = T extends Promise<infer V> ? V : T
+type ResolveAtom<T> = T extends Atom<infer V> ? V : T
+type ResolveType<T> = ResolvePromise<ResolveAtom<T>>
 
 export function waitForAll<
-  Values extends Record<string, unknown> | readonly unknown[]
->(atoms: { [K in keyof Values]: Atom<Values[K]> }): Atom<{
+  Values extends Record<string, Atom<unknown>> | readonly Atom<unknown>[]
```

Comment:
```suggestion
  Atoms extends Record<string, Atom<unknown>> | readonly Atom<unknown>[]
```
Can we rename like this?

Reply:
👍